packages/core/src/resource/resource.ts

```
@@ -162,7 +168,19 @@ export class ResourceImpl<T, R> extends BaseWritableResource<T> implements Resou
       computed(
         () => {
           const streamValue = this.state().stream?.();
-          return streamValue && isResolved(streamValue) ? streamValue.value : this.defaultValue;
+
+          if (!streamValue) {
+            return defaultValue;
+          }
+
+          if (!isResolved(streamValue)) {
```

If the resource is in the error state, we should throw the error the loader gave us, from the error signal, not a new `RuntimeError`.

---

Done